### PR TITLE
SUSPENDED subscriptions can be renewed

### PR DESCRIPTION
**Pull request by jarshwah**

### CODE REVIEW

**Comment by jarshwah** [comment]
[Fix double billing resulting from Search Alert: Kogan First Billing Couldn't Be Setup](https://trello.com/c/omta0dqf/9131-fix-double-billing-resulting-from-search-alert-kogan-first-billing-couldnt-be-setup)

File: src/subscriptions/models.py [hunk 1]
```diff
@@ -190,7 +190,9 @@ def post_renew(self):
 
     @fsm_log_description
     @transition(
-        field=state, source=[State.ACTIVE, State.RENEWING, State.ERROR], target=State.ACTIVE
+        field=state,
+        source=[State.ACTIVE, State.RENEWING, State.SUSPENDED, State.ERROR],
+        target=State.ACTIVE,
```
Comment (DaveDM) [inline src/subscriptions/models.py:195]: NIT: Your diagram is getting out of date.

**Review by DaveDM — COMMENTED** [review]
Seems reasonable, but kind-of overloads the definition of what suspended means.

It's merged already, so this question is mostly academic, but did you consider having a separate state of a subscription for a timeout?

With a timeout, the subscription renewal hasn't necessarily failed, it's just also not suspended.

For these Finite State machines, do you prefer having a small number of states, or fewer loops?

**Comment by jarshwah** [comment]
You're absolutely right - and we have an error state for exactly that purpose `state_unknown` which *can* move back to renewed when we receive new information. I'll probably revert this change.

**Pull request merged**